### PR TITLE
Change Email from Profile page

### PR DESCRIPTION
**Pull request by habdelra**
This PR introduces the ability to change your email from the profile page.

There are a couple quirks that the Matrix API forces us to deal with:
1. the matrix API requires us to provide the user's password for additional security when changing a user's email address
2. the matrix API always sends the verification email first _before_ verifying that the password provided is actually correct

change-email

### CODE REVIEW

File: packages/host/app/components/operator-mode/profile-settings-modal.gts [hunk 1]
```diff
@@ -1,128 +1,137 @@
...
+        </div>
+      </:footer>
+    </ModalContainer>
     <style>
+      .password-modal :deep(.boxel-modal__inner) {
+        height: 21rem;
+        margin-top: calc((100vh - 21rem) / 2);
+      }
+      .password-modal :deep(.invalid) {
+        box-shadow: none;
```
Comment (habdelra) [inline packages/host/app/components/operator-mode/profile-settings-modal.gts]: there seems to be a box shadow now on the invalid indicator that looks really bad with the `BoxelInput` component. perhaps this was an unintentional side effect? I'm turning it off everywhere I'm encountering it

File: packages/host/app/components/operator-mode/profile-settings-modal.gts [hunk 2]
```diff
@@ -160,11 +169,149 @@ export default class ProfileSettingsModal extends Component<Signature> {
...
+                    </p>
+                  </div>
+                {{/if}}
+              </div>
+            </FieldContainer>
+          {{/if}}
         </form>
 
-        {{#if this.error}}
+        {{! TODO this should move into the display field input}}
```
Comment (habdelra) [inline packages/host/app/components/operator-mode/profile-settings-modal.gts]: don't forget to do this...

File: packages/host/app/components/operator-mode/profile-settings-modal.gts [hunk 3]
```diff
@@ -160,11 +169,126 @@ export default class ProfileSettingsModal extends Component<Signature> {
               }}
             />
           </FieldContainer>
+          {{#if this.hasPendingEmailChange}}
+            <FieldContainer
+              @label='Current Email'
+              @tag='label'
+              class='profile-field'
+              @vertical={{false}}
```
Comment (jurgenwerk) [inline packages/host/app/components/operator-mode/profile-settings-modal.gts]: Can `@vertical={{false}}` be removed?

**Review by jurgenwerk — APPROVED** [review]
Looks great! I think all those `vertical={{false}}` can be removed because it looks like this is the default for FieldContainer

File: packages/matrix/docker/synapse/index.ts [hunk 4]
```diff
@@ -307,3 +309,48 @@ export async function createRegistrationToken(
...
+    emailAddresses,
+  }: {
+    password?: string;
+    displayname?: string;
+    avatar_url?: string;
+    emailAddresses?: string[];
+  },
+) {
+  let res = await fetch(
+    `http://localhost:${synapse.mappedPort}/_synapse/admin/v2/users/${userId}`,
```
Comment (lukemelia) [inline packages/matrix/docker/synapse/index.ts:330]: Any reason we use localhost here but 127.0.0.1 on line 240 above?

**Review by lukemelia — CHANGES_REQUESTED** [review]
The initial state of the settings screen should not show the New email field. It should be like this: https://app.zeplin.io/project/652ea0fee5ade422add72a63/screen/652ea4648d18982039bee37d

**Review by burieberry — APPROVED** [review]

File: packages/matrix/docker/synapse/index.ts (hunk 4 above)
Comment (habdelra) [reply packages/matrix/docker/synapse/index.ts:330]: no reason, i think it was just copy-pasta

**Comment by habdelra** [comment]
> The initial state of the settings screen should not show the New email field. It should be like this: https://app.zeplin.io/project/652ea0fee5ade422add72a63/screen/652ea4648d18982039bee37d

ah, i see. i never saw that screen in the linear ticket

**Comment by habdelra** [comment]
@lukemelia I've updated the flow to look like this now:
change-email-update

**Review by lukemelia — APPROVED** [review]

**Comment by lukemelia** [comment]
Looks good. Chris may have tweaks but I think we should get this merged and we can review with him then.

**Pull request merged**